tendrl/commons/flows/expand_cluster_with_detected_peers/__init__.py

```diff
@@ -0,0 +1,144 @@
+            raise FlowExecutionFailedError(
+                "Another job in progress for cluster, please wait till "
+                "the job finishes (job_id: %s) (integration_id: %s) " % (
+                    _cluster.current_job['job_id'],
+                    integration_id
+                )
+            )
+        _lock_details = {
+            'node_id': NS.node_context.node_id,
+            'fqdn': NS.node_context.fqdn,
+            'tags': NS.node_context.tags,
+            'type': NS.type,
+            'job_name': self.__class__.__name__,
+            'job_id': self.job_id
+        }
+        _cluster.locked_by = _lock_details
+        _cluster.status = "expanding"
+        _cluster.current_job = {
+            'job_id': self.job_id,
+            'job_name': self.__class__.__name__,
+            'status': 'in_progress'
+        }
+        _cluster.save()
+
+        try:
+            integration_id_index_key = \
+                "indexes/tags/tendrl/integration/%s" % integration_id
+            node_ids = etcd_utils.read(
+                integration_id_index_key).value
+            node_ids = json.loads(node_ids)
+        except etcd.EtcdKeyNotFound:
+            raise FlowExecutionFailedError("Cluster with "
+                                           "integration_id "
+                                           "(%s) not found, cannot "
+                                           "import" % integration_id)
+
+        job_ids = []
+        for node_id in node_ids:
+            _cnc = NS.tendrl.objects.ClusterNodeContext(
+                node_id=node_id
+            ).load()
+            if _cnc.is_managed.lower() == "yes":
+                continue
+
+            params = {
+                'TendrlContext.integration_id': integration_id,
+                'Node[]': [node_id],
+                'Cluster.enable_volume_profiling':
+                _cluster.enable_volume_profiling
+            }
+            payload = {
+                "tags": ["tendrl/node_%s" % node_id],
+                "run": "tendrl.flows.ImportCluster",
+                "status": "new",
+                "parent": self.parameters['job_id'],
+                "parameters": params,
+                "type": "node"
+            }
+            _job_id = str(uuid.uuid4())
+            Job(job_id=_job_id, status="new", payload=payload).save()
+            logger.log(
+                "info",
+                NS.publisher_id,
+                {
+                    "message": "Importing (job: %s) Node %s "
+                    "to cluster %s" % (
+                        _job_id,
+                        node_id,
+                        integration_id
+                    )
+                },
+                job_id=self.parameters['job_id']
+            )
+            job_ids.append(_job_id)
+
+        loop_count = 0
+        # Wait for (no of nodes) * 6 minutes for import to complete
+        wait_count = len(node_ids) * 36
```

Comment:
@r0h4n `node_ids` is not the list of nodes with which the cluster is expanded. Rather this is the full list of nodes as few skips might happen at L#70. We should wait for only no of nodes which actually executed import cluster flow here.

Reply:
please send a fix